src/terraform.yml

```
@@ -39,14 +39,16 @@ repository:
       - include: "#expressions"
 
   functions:
-    begin: (\w+)(\()
+    begin: ([:\-\w]+)(\()
     name: meta.function-call.hcl
     comment: Built-in function calls
     beginCaptures:
       "1":
         patterns:
           - match: \b(abs|abspath|alltrue|anytrue|base64decode|base64encode|base64gzip|base64sha256|base64sha512|basename|bcrypt|can|ceil|chomp|chunklist|cidrhost|cidrnetmask|cidrsubnet|cidrsubnets|coalesce|coalescelist|compact|concat|contains|csvdecode|dirname|distinct|element|endswith|file|filebase64|filebase64sha256|filebase64sha512|fileexists|filemd5|fileset|filesha1|filesha256|filesha512|flatten|floor|format|formatdate|formatlist|indent|index|join|jsondecode|jsonencode|keys|length|log|lookup|lower|matchkeys|max|md5|merge|min|nonsensitive|one|parseint|pathexpand|plantimestamp|pow|range|regex|regexall|replace|reverse|rsadecrypt|sensitive|setintersection|setproduct|setsubtract|setunion|sha1|sha256|sha512|signum|slice|sort|split|startswith|strcontains|strrev|substr|sum|templatefile|textdecodebase64|textencodebase64|timeadd|timecmp|timestamp|title|tobool|tolist|tomap|tonumber|toset|tostring|transpose|trim|trimprefix|trimspace|trimsuffix|try|upper|urlencode|uuid|uuidv5|values|yamldecode|yamlencode|zipmap)\b
             name: support.function.builtin.terraform
+          - match: \bprovider::[^\d\W][\w_-]*::[^\d\W][\w_-]*\b
```

```suggestion
          - match: \bprovider::[[:alpha:]][\w_-]*::[[:alpha:]][\w_-]*\b
```
Stating what's allowed here instead of what's not allowed makes it more readable IMHO. What do you think?

---

Yeah, I agree – the only reason I ended with that other expression was because I removed `[[:alpha:]]` in the first place 😂